src/main/java/de/neuland/jade4j/model/JadeModel.java

```diff
@@ -22,6 +25,10 @@
 	private Map<String, MixinNode> mixins = new HashMap<String, MixinNode>();
 	private Map<String, Filter> filter = new HashMap<String, Filter>();
 
+	// to detect infinite loop
+	private Map<String, Integer> visited = new HashMap<String, Integer>();
```

Comment:
Are you sure this class (JadeModel) and therefore this Map will be newly instantiated every time we parse a different document?

Reply:
Could add a unit test to prove this by parsing the same document multiple times.